Add auto open in browser for webapp

sztang opened the pull request:
Automatically open localhost for webapp, unless ioloop = None and run_until_shutdown() is called. Also added missing import from bokeh.io for showing jupyter notebook.

verybadsoldier reviewed (commented):
Thanks for the PR. I added some comments.

sztang commented on backtrader_plotting/bokeh/bokeh_webapp.py:
```diff
@@ -50,11 +52,13 @@ def _run_server(fnc_make_document, iplot=True, notebook_url="localhost:8889", po
             server = Server(apps, port=port, io_loop=ioloop)
             if ioloop is None:
                 server.run_until_shutdown()
+                webbrowser.open(f'http://localhost:{port}')
```
Yep, realized the problem and adjusted in second commit.

verybadsoldier replied:
Oh right, sorry, my bad. Could you please squash the 2 commits and make it a single commit? You can just do that in your branch. GitHub will automatically update the PR.

sztang commented on demos/optimization.py:
```diff
@@ -30,7 +30,7 @@ def next(self):
 
     data = bt.feeds.YahooFinanceCSVData(
-        dataname="datas/orcl-1995-2014.txt",
+        dataname="demos/datas/orcl-1995-2014.txt",
```
When running locally on vscode the filepath that YahooFinanceCSVData() used was the root directory instead of the demos directory though, and that's why I had to change this to find the file. Is that something of concern?

Another thing is that while running locally the edits in bokeh_webapp.py don't show up - I tried commenting out the print(f"Open your browser here: http://localhost:{port}") statement but it didn't change the output. At first I thought it might be a module import caching issue but couldn't fix with reload/refreshing. Any idea what's going on? Thanks!

verybadsoldier replied:
1.) I don't use vscode but I imagine you have to set the working directory of your run configuration to the `demos` directory. I guess otherwise it will default to the project's root directory.

2.) I would try stopping in the code in th debugger. You should be able to see the full file path it actually is running on. Maybe you installed the package and you are running the code from the regular `site-packages` directory instead of your checkout directory?

sztang commented:
@verybadsoldier resolved the issue with testing, the imports were indeed from the module and not from the directory's files. Made the changes, tested and reverted. In the process I've also added threading.Timer to open the webbrowser with a 2 second delay, so it should work for both the run_until_shutdown server and the normal server. Squashed everything into 1 commit too. Thoughts?

verybadsoldier commented:
Thanks!
